src/codecatalyst/devEnv.ts

```diff
@@ -0,0 +1,206 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DevEnvironment } from '../shared/clients/codecatalystClient'
+import { DevEnvClient } from '../shared/clients/devenvClient'
+import globals from '../shared/extensionGlobals'
+import * as vscode from 'vscode'
+import { Timeout } from '../shared/utilities/timeoutUtils'
+import { showMessageWithCancel } from '../shared/utilities/messages'
+
+/** If we should be sending the dev env heartbeats to track user activity */
+export function shouldTrackUserActivity(maxInactivityMinutes: DevEnvironment['inactivityTimeoutMinutes']): boolean {
+    // This value is a static value initialized when the dev env is first created, if it is updated the dev env is restarted
+    // For more info see: https://docs.aws.amazon.com/codecatalyst/latest/APIReference/API_UpdateDevEnvironment.html#codecatalyst-UpdateDevEnvironment-request-inactivityTimeoutMinutes
+    return maxInactivityMinutes > 0
+}
+
+export class InactivityMessages {
+    static #instance: InactivityMessages | undefined
+
+    static get instance(): InactivityMessages {
+        return (this.#instance ??= new InactivityMessages())
+    }
+
+    /**
+     * Sets up all messages that are displayed to the user when the dev env is inactive and starts to get close to shutdown.
+     *
+     * @param inactivityMinutes The # of minutes a CodeCatalyst Dev Env can be inactive till it shuts down.
+     * @param devEnv The DevEnv client
+     * @returns
+     */
+    async setupMessages(
+        inactivityMinutes: DevEnvironment['inactivityTimeoutMinutes'],
+        devEnv: DevEnvClient,
+        warningMessage: InactivityWarningMessage,
+        shutdownMessage: InactivityShutdownMessage
+    ) {
+        const maxInactivityMinutes = inactivityMinutes
```

Comment:
Perhaps rename `inactivityMinutes` instead

Reply:
Renamed the param to `maxInactivityMinutes` instead